### PR TITLE
docs(proposals): add How? for dynamic log level admin API (#798)

### PR DESCRIPTION
**Pull request by henschwartz**
## Summary
- Add Decisions closing the six Open Questions from the #798 What?/Why? proposal
- Add How? design for runtime `EnvFilter` reload via admin `PUT`/`GET`/`HEAD` `/api/log-level`
Follows merged What?/Why? in #948.
## Test plan
- [ ] Proposal frontmatter and issue links valid
- [ ] Every Open Question has a matching Decisions bullet
- [ ] How? field contracts list concrete defaults (e.g. duration_secs 300, cap 86400)

### CODE REVIEW

File: docs/proposals/00798_dynamic-log-level-admin-api.md [hunk 1]
```diff
@@ -235,3 +235,157 @@ they are not separate tracked issues.
...
+validation matches config validation.
+
+**Response errors.** **400** for invalid JSON, unknown fields,
+bad level/module/duration. **405** for wrong method. Serialization
+failures log with `tracing::error!` then return **500** (same
+pattern as `pipelines_admin::serialization_failed_response`).
+
+**Reload interaction.** After successful `reload_pipelines`, call
+into log-level state to refresh `baseline_directive` from the new
+config without clearing overlays (Decisions above).
```
[the remark this answers is not in the document]
Comment (henschwartz) [reply docs/proposals/00798_dynamic-log-level-admin-api.md]: Fixed: documented Arc<LogLevelState> with a tokio::sync::Mutex guarding the full read-overlay / rebuild-filter / reload() sequence for PUT, DELETE, timer revert, and post-reload baseline refresh.

File: docs/proposals/00798_dynamic-log-level-admin-api.md [hunk 2]
```diff
@@ -235,3 +235,157 @@ they are not separate tracked issues.
...
+**Admin HTTP surface.** Add `log_level_admin` beside
+`pipelines_admin` (`protocol/src/http/pingora/health/`). Route in
+`PingoraAdminService::response` when `path == "/api/log-level"`:
+
+| Method | Behavior |
+| --- | --- |
+| `PUT` | Parse JSON body; validate; apply overlay; schedule revert; reload filter; **200** with current state body |
+| `GET` | Return structured state JSON (**200**) |
+| `HEAD` | Same as `GET` but strip body per `as_head_response` (RFC 9110 §8.6) |
+| other | **405** with `Allow: GET, HEAD, PUT` |
```
[the remark this answers is not in the document]
Comment (henschwartz) [reply docs/proposals/00798_dynamic-log-level-admin-api.md]: Fixed: added DELETE /api/log-level with optional module query and ?all=true to clear overlays before timer expiry; updated Allow header and graduation criteria.

File: docs/proposals/00798_dynamic-log-level-admin-api.md [hunk 3]
```diff
@@ -235,3 +235,157 @@ they are not separate tracked issues.
...
+
+**Request body (`PUT`).**
+
+| Field | Contract |
+| --- | --- |
+| `level` | required; one of `error`, `warn`, `info`, `debug`, `trace` (case-insensitive) |
+| `module` | optional; when set, per-module overlay; when omitted, global overlay |
```
[the remark this answers is not in the document]
Comment (henschwartz) [reply docs/proposals/00798_dynamic-log-level-admin-api.md]: Fixed: contract now explicitly rejects module: "" with 400; empty string cannot alias the global overlay key.

File: docs/proposals/00798_dynamic-log-level-admin-api.md [hunk 4]
```diff
@@ -235,3 +235,157 @@ they are not separate tracked issues.
...
+| `PUT` | Parse JSON body; validate; apply overlay; schedule revert; reload filter; **200** with current state body |
+| `GET` | Return structured state JSON (**200**) |
+| `HEAD` | Same as `GET` but strip body per `as_head_response` (RFC 9110 §8.6) |
+| other | **405** with `Allow: GET, HEAD, PUT` |
+
+**Request body (`PUT`).**
+
+| Field | Contract |
+| --- | --- |
+| `level` | required; one of `error`, `warn`, `info`, `debug`, `trace` (case-insensitive) |
```
[the remark this answers is not in the document]
Comment (henschwartz) [reply docs/proposals/00798_dynamic-log-level-admin-api.md]: Fixed: admin PUT accepts off via is_valid_admin_log_level; static log_overrides / is_valid_log_level unchanged.

File: docs/proposals/00798_dynamic-log-level-admin-api.md [hunk 5]
```diff
@@ -235,3 +235,157 @@ they are not separate tracked issues.
...
+into log-level state to refresh `baseline_directive` from the new
+config without clearing overlays (Decisions above).
+
+**Tests.**
+
+- Unit: directive rebuild (baseline only, baseline + overlays,
+  replace timer), validation edge cases (`duration_secs` 0 / 86_401)
+- Integration: start proxy with `log_overrides`; `PUT` raises a
+  module to `trace`; assert a log line at the new level; wait (or
+  mock clock) for revert; assert level drops back
```
[the remark this answers is not in the document]
Comment (henschwartz) [reply docs/proposals/00798_dynamic-log-level-admin-api.md]: Fixed: integration test plan commits to tokio::time::pause() and tokio::time::advance() instead of wall-clock sleep; includes DELETE-before-expiry coverage.

**Review by shaneutt — APPROVED** [review]
